### PR TITLE
#527 Implements retries while fetching the wikipedia extracts. 

### PR DESCRIPTION
**Pull request by SpaceDave42**
#527 Each time a language and type extract fails the script tries again with a smaller batch size. This also changes request_utils.py as this script handled the error caused by the chunk size being too large by calling exit(-1), completely stopping the crawl which is less than ideal.

This has been tested in branch 524 and is working as intended.

### CODE REVIEW

File: etl/data_extraction/get_wikipedia_extracts.py [hunk 1]
```diff
@@ -213,39 +213,60 @@ def add_wikipedia_extracts(language_keys: Optional[List[str]] = lang_keys, ) ->
...
+                                )
+
+                            # set done to true after all items have been processed
+                            done = True
+                        except Exception as error:
+                            logger.error(f"Fetching wikipedia extracs for {filename}, lang:{key} and chunk size:{chunk_size} failed!")
+                            logger.error(error)
+
+                            # lower chunk size and try again in while loop
+                            chunk_size -= 5
```
Comment (EeveesEyes) [inline etl/data_extraction/get_wikipedia_extracts.py:263]: I think we should increase the chunk_size again after successful fetches, but no further than 20. 
When one of the first chunks fails and the chunk size is lowered to 5, the rest of the fetching will take ages.

**Review by EeveesEyes — CHANGES_REQUESTED** [review]
Consider my comment - other than that, this is ready to go :+1:

File: etl/data_extraction/get_wikipedia_extracts.py (hunk 1 above)
Comment (SpaceDave42) [reply etl/data_extraction/get_wikipedia_extracts.py:267]: Good idea

**Comment by SpaceDave42** [comment]
I pushed an increase to the chunk size after a successful chunk as long as it it <20.

**Review by EeveesEyes — APPROVED** [review]
:ok_hand:

**Pull request merged**